packages/examples/cvat/exchange-oracle/tests/api/test_exchange_api.py

```diff
@@ -1,100 +1,306 @@
 import json
+import math
 import uuid
 from datetime import datetime, timedelta
+from itertools import combinations
+from time import sleep
 from unittest.mock import patch
 
+from fastapi.responses import Response
 from fastapi.testclient import TestClient
+from jose import jwt
 
-from src.core.types import AssignmentStatuses
+from src.core.config import Config
+from src.core.types import AssignmentStatuses, JobStatuses, ProjectStatuses
 from src.db import SessionLocal
-from src.models.cvat import Assignment, User
-
-from tests.utils.db_helper import create_project_task_and_job
+from src.models.cvat import Assignment, Project, User
+from src.schemas.exchange import AssignmentStatuses as APIAssignmentStatuses
+from src.schemas.exchange import JobStatuses as APIJobStatuses
+
+from tests.utils.db_helper import (
+    create_job,
+    create_project,
+    create_project_and_task,
+    create_project_task_and_job,
+    create_task,
+)
 
 escrow_address = "0x12E66A452f95bff49eD5a30b0d06Ebc37C5A94B6"
 user_address = "0x86e83d346041E8806e352681f3F14549C0d2BC60"
 cvat_email = "test@hmt.ai"
 
 
-def test_empty_list_tasks_200_with_address(client: TestClient) -> None:
+def generate_jwt_token(
+    *,
+    wallet_address: str | None = user_address,
+    email: str = cvat_email,
+) -> str:
+    return jwt.encode(
+        {
+            **({"wallet_address": wallet_address} if wallet_address else {"role": "HUMAN_APP"}),
+            "email": email,
+        },
+        Config.human_app_config.jwt_key,
+    )
+
+
+def get_auth_header(token: str = generate_jwt_token()) -> dict:
+    return {"Authorization": f"Bearer {token}"}
+
+
+empty_result = {
+    "page": 0,
+    "total_pages": 0,
+    "total_results": 0,
+    "results": [],
+    "page_size": 5,  # default page size
+}
+
+
+def test_empty_list_jobs_200_with_address(client: TestClient) -> None:
```

Comment:
Please rename tests to use pattern "test_can_*". It makes it more clear what is the expected outcome in the case being tested.

Reply:
Done